topics/topic01/morley.js

```diff
@@ -1,94 +1,164 @@
-        a2 = flag * a2;
-        v3 = v3.rotate(-a2 / 3);
-
-        return getLineIntersection(B, v1, C, v3);
-    }
-
-    pA = [423, 360];
-    pB = [76, 360];
-    letterC.setAttribute("x", pA[0] + 5);
-    letterC.setAttribute("y", pA[1] + 5);
-    letterB.setAttribute("x", pB[0] - 15);
-    letterB.setAttribute("y", pB[1] + 5);
-    const morley = (p) => {
-        let pC = [p.x, p.y];
-        p = [166, 51];
-        let points = [pA, pB, pC];
-        tri1.setAttributeNS(null, "points", makeString(points));
-
-        letterA.setAttribute("x", pC[0]);
-        letterA.setAttribute("y", pC[1] - 5);
-        let A = new Vector(pA[0], pA[1]);
-        let B = new Vector(pB[0], pB[1]);
-        let C = new Vector(pC[0], pC[1]);
-
-        if (pC[1] > pA[1]) {
-            flag = -1;
-        } else {
-            flag = 1;
-        }
-
-        let Cprime = GetD(C, A, B, flag);
-        let Bprime = GetD(B, C, A, flag);
-        let Aprime = GetD(A, B, C, flag);
-        let cP = [Cprime.x, Cprime.y];
-        let bP = [Bprime.x, Bprime.y];
-        let aP = [Aprime.x, Aprime.y];
-        drawLine(cP, pB, line1);
-        drawLine(aP, pB, line2);
-        drawLine(cP, pA, line3);
-        drawLine(bP, pA, line4);
-        drawLine(aP, pC, line5);
-        drawLine(bP, pC, line6);
-        letterAp.setAttribute("x", cP[0]);
-        letterAp.setAttribute("y", cP[1]);
-        letterBp.setAttribute("x", bP[0]);
-        letterBp.setAttribute("y", bP[1]);
-        letterCp.setAttribute("x", aP[0]);
-        letterCp.setAttribute("y", aP[1]);
-        let primePoints = [
-            [Cprime.x, Cprime.y],
-            [Bprime.x, Bprime.y],
-            [Aprime.x, Aprime.y]
-        ];
-        tri2.setAttributeNS(null, "points", makeString(primePoints));
-    };
-
-    morley(new Vector(166, 51));
-
-    svg.addEventListener("mousemove", (event) => {
-        var p = new DOMPoint(event.clientX, event.clientY);
-
-        p = p.matrixTransform(svg.getScreenCTM().inverse());
-
-        morley(p);
+    const canvas = new fabric.Canvas("morley-canvas", {
+        selection: false
     });
-}
\ No newline at end of file
+
+    const ABprime = makeLine(null,null,null,"purple");
+    const ACprime = makeLine(null,null,null,"purple");
+    const BCprime = makeLine(null,null,null,"purple");
+    const BAprime = makeLine(null,null,null,"purple");
+    const CAprime = makeLine(null,null,null,"purple");
+    const CBprime = makeLine(null,null,null,"purple");
+    const AprimeBprime = makeLine(null,null,null,"red");
+    const BprimeCprime= makeLine(null,null,null,"red");
+    const CprimeAprime = makeLine(null,null,null,"red");
+
+
+
+    // vertexes
+    const aLabel = makeLabel("A");
+    const bLabel = makeLabel("B");
+    const cLabel = makeLabel("C");
+    const aprimeLabel = makeLabel("A'",18);
+    const bprimeLabel = makeLabel("B'",18);
+    const cprimeLabel = makeLabel("C'",18);
+
+
+    const triangle = makeMovablePolygon([{
```

Comment:
Does it look like this triangle is hardcoded?  

Are we suppose to pass the value  coords to this triangle function ?   I can't find the value is been set anywhere ... 

usually it is set through mousemove event listener...  svg.addEventListener("mousemove", (event) => {